app/models/user.rb

```diff
@@ -1,18 +1,29 @@
 class User < ApplicationRecord
+  include ::Exceptions::User
+
   validates :name, presence: true, uniqueness: true
 
   # Validates that the two passwords match in field
   has_secure_password
 
+  before_destroy :ensure_not_admin
+  
   ## Creating Transaction/Trigger that will rollback when last user deleted
   after_destroy :ensure_an_admin_remains
 
-  class Error < StandardError
+  after_create_commit do |user|
+    UserMailer.welcome(user).deliver_now
   end
 
+  before_update :ensure_not_admin
+
   private def ensure_an_admin_remains
     if User.count.zero?
       raise Error.new "Cant't delete last user"
     end
   end
+
+  private def ensure_not_admin
+    raise AdminImmutableException if email_was == 'admin@depot.com'
```

Comment:
1. There should be separate method to check if the user is admin
2. Instead of raising exceptions add errors